crest/Assets/Crest/Crest/Scripts/LodData/RegisterHeightInput.cs

```diff
@@ -37,10 +39,58 @@ public partial class RegisterHeightInput : RegisterLodDataInputWithSplineSupport
 
         protected override bool FollowHorizontalMotion => true;
 
-        [Header("Height Input Settings")]
```

Comment:
Predicated breaking the foldout is a bug. I know how to fix it though. I can also update Predicated to completely hide fields instead of just disabling.

As for the Header not disappearing, I will have to look at that too.

A question will be whether from a code perspective we'd want to group fields into classes or not. If not I can look at doing a foldout attribute. But maybe it makes sense to do so.

Also there was a comment about lambda not working with Predicated? I cannot find it anymore, but it does support calling methods. `Predicated(Type type, string methodName)`

I think reworking the UI will be easier in separate PR if that's okay. Keep it simple for now. I'm not in favour of the current header style. Just normal headings should be ok. But I will leave up to you whether to keep it.

Reply:
thanks. yeah UI rework is fine in separate PR, and can be post 2022 depending on how things go. could you add a trello for it and stick it in somewhere?

i've used the fix you pushed. theres the slight issue that some of the spline settings want to be predicated on multiple things (both 'override=true' and 'mode=spline'). it works ok the way i've done it, i think. rather than supporting predicating on multiple fields, i wonder if having an additional type of predication that completely hides the param rather than disabling is better, ideally the UI would always only show relevant stuff. but that can be a discussion for later.

Reply:
Sure. Done.

I agree. I can get around to that as part of the UI rework.